### ISSUE TITLE
TensorFlow op to copy weights of Keras model

### ISSUE THREAD

**Issue by volvador** [tensorflow/tensorflow#16584]
I am doing a distributed calibration of an LSTM model (keras 2.0 + TensorFlow 1.0)

    with tf.device(tf.train.replica_device_setter(...):
          model = ##create model by keras
          clone_model = ## create the same model by keras but now a stateful one

after calibration, I want my chief worker to use the clone_model, copy the weights the calibration reached in model, and make predictions on some test set, but simply calling

     clone_model.set_weights(model.get_weights())

does not work.
I understand I need to define this weight copy as an op and then call session(run) of that op

Can you please help with a TensorFlow op copying weights of a keras model to another (identical architecture) Keras model?

**Comment by volvador** [comment]
Thank you. I think it is `trainable_weights` instead of `trainable_variables` 
Inspired by your answer this is what I did 

    clone_ops = []
    for i, src in enumerate(model.trainable_weights):
          op = tf.assign(clone_model.trainable_weights[i], src)
          clone_ops.append(op)

and then when I want to clone the weights

    for op in clone_ops:
        session.run(op)

**Comment by drpngx** [comment]
Glad you figured it out!
